chain/network/src/concurrency/arc_mutex.rs

```diff
@@ -12,16 +12,32 @@ impl<T: Clone> ArcMutex<T> {
     pub fn new(v: T) -> Self {
         Self { value: ArcSwap::new(Arc::new(v)), mutex: Mutex::new(()) }
     }
-    // non-blocking
+
+    /// Loads the last value stored. Non-blocking.
     pub fn load(&self) -> Arc<T> {
         self.value.load_full()
     }
-    // blocking
-    pub fn update<R>(&self, f: impl FnOnce(&mut T) -> R) -> R {
+
+    /// Atomic update of the value. Blocking.
+    /// Note that `T -> (R,T)` is a state monad.
```

Comment:
I'd add more info (as "state monad" doesn't say much for people who didn't use haskell in the past).

Reply:
done